core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -93,10 +94,12 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   private val NUM_PROCESSING_THREADS = conf.getInt(SPARK_HISTORY_FS_NUM_REPLAY_THREADS,
     Math.ceil(Runtime.getRuntime.availableProcessors() / 4f).toInt)
 
-  private val logDir = conf.getOption("spark.history.fs.logDirectory")
+  private val logURIString = conf.getOption("spark.history.fs.logDirectory")
     .map { d => Utils.resolveURI(d).toString }
     .getOrElse(DEFAULT_LOG_DIR)
 
+  private val logDir = URLDecoder.decode(logURIString, "UTF-8")
```

Comment:
Hm, this is either a local path, in which case it shouldn't be URI-escaped, or it's a URI, in which case it can be parsed with `java.net.URI` right? I think we take the latter approach in most other places in the code.